src/CADL.Extension/Emitter.Csharp/test/Unit/usage.test.ts

```diff
@@ -0,0 +1,256 @@
+                name: string;
+            }
+            op test(@path id: string, @body foo: Foo): void;
+            op test2(@path id: string): BaseModel;
+      `,
+            runner
+        );
+        const context = createEmitterContext(program);
+        const dpgContext = createDpgContext(context);
+        const [services] = getAllHttpServices(program);
+        const usages = getUsages(dpgContext, services[0].operations);
+        // verify that the baseModel will not apply the usage of derived model.
+        assert(usages.outputs.includes("BaseModel"));
+        // verify that the derived model will inherit the usage of base model
+        assert(usages.roundTrips.includes("Foo"));
+    });
+
+    it("Test the usage of models spread alias", async () => {
+        const program = await typeSpecCompile(
+            `
+            alias FooAlias = {
+                @path id: string;
+                @doc("name of the Foo")
+                name: string;
+            };
+            op test(...FooAlias): void;
+      `,
+            runner
+        );
+        const context = createEmitterContext(program);
+        const dpgContext = createDpgContext(context);
+        const [services] = getAllHttpServices(program);
+        const usages = getUsages(dpgContext, services[0].operations);
+        assert(usages.inputs.includes("TestRequest"));
+    });
+
+    it("Test the usage of body parameter of azure core operation.", async () => {
+        const program = await typeSpecCompile(
+            `
+            @doc("This is a model.")
+            @resource("items")
+            model Foo {
+                @doc("id of Foo")
+                @key
+                @visibility("read","create","query")
+                id: string;
+                @doc("name of Foo")
+                name: string;
+            }
+
+            @doc("The item information.")
+            model FooInfo {
+                @doc("name of Foo")
+                name: string;
+            }
+
+            @doc("this is a response model.")
+            model BatchCreateFooListItemsRequest {
+                @doc("The items to create")
+                fooInfos: FooInfo[];
+            }
+
+            @doc("this is a response model.")
+            model BatchCreateTextListItemsResponse {
+                @doc("The item list.")
+                fooList: Foo[];
+            }
+            interface TextLists{
+                @doc("create items")
+                addItems is ResourceAction<Foo, BatchCreateFooListItemsRequest, BatchCreateTextListItemsResponse>;
+            }
+      `,
+            runner,
+            true,
+            true
+        );
+        const context = createEmitterContext(program);
+        const dpgContext = createDpgContext(context);
+        const [services] = getAllHttpServices(program);
+        const usages = getUsages(dpgContext, services[0].operations);
+        assert(usages.inputs.includes("BatchCreateFooListItemsRequest"));
```

Comment:
Is it beeter to add `usages.roundtrips.includes("BatchCreateFooListItemsRequest") === false` ?

Same to the other tests. If this is to fix the mismatch issue, we should make sure the models not in the inappropriate place.

Reply:
Each model only occur in one array (inputs/outputs/roundtrips), so it will not need to check it is not in other array.